scenes/game/local_game.gd

```diff
@@ -4708,6 +4906,163 @@ func handle_strike_effect(card_id :int, effect, performing_player : Player):
 				_append_log_full(Enums.LogType.LogType_CharacterMovement, performing_player, "moves %s to themselves on space %s." % [buddy_name, str(space)])
 			else:
 				_append_log_full(Enums.LogType.LogType_CharacterMovement, performing_player, "moves %s to themselves, from space %s to %s." % [buddy_name, str(old_buddy_pos), str(space)])
+		"place_next_buddy":
+			var require_unoccupied = effect['require_unoccupied']
+			var destination = effect['destination']
+			var num_buddies = effect['amount']
+			var valid_new_positions = [1,2,3,4,5,6,7,8,9]
+			var already_removed_buddy = 'already_removed_buddy' in effect and effect['already_removed_buddy']
+			if already_removed_buddy:
+				valid_new_positions = effect['valid_new_positions']
+			else:
+				# Filter based on destination requirements.
+				match destination:
+					"attack_range":
+						var attack_card = active_strike.get_player_card(performing_player)
+						for i in range(valid_new_positions.size() - 1, -1, -1):
+							var check_position = valid_new_positions[i]
+							if not is_location_in_range(performing_player, attack_card, check_position):
+								valid_new_positions.remove_at(i)
+					"anywhere":
+						pass
+					"adjacent_self":
+						for i in range(valid_new_positions.size() - 1, -1, -1):
+							var check_position = valid_new_positions[i]
+							if not (check_position == performing_player.arena_location - 1 or check_position == performing_player.arena_location + 1):
```

Comment:
even if there's no wide card that would use this i'd be more comfortable minimizing where we use the arena_location variable like this; I prefer what you did earlier with checking is_in_location for the spaces next to the potential spike location